pvlib/iotools/solrad.py

```diff
@@ -91,19 +100,28 @@ def read_solrad(filename):
         widths = WIDTHS
         dtypes = DTYPES
 
+    meta = {}
+
+    if str(filename).startswith('ftp') or str(filename).startswith('http'):
+        response = requests.get(filename)
+        file_buffer = io.StringIO(response.content.decode())
+    else:
+        file_buffer = open(str(filename), 'r')
```

Comment:
```suggestion
        with open(str(filename), 'r') as f:
            file_buffer = io.StringIO(f.read())
```
This would be an easy-to-follow allergy suppression. Faster too.

Reply:
Thanks @adriesse 😄